internal/cmd/gencli/templates.go

```diff
@@ -385,10 +385,11 @@ func (c *{{ camelCase .SubActionPrefix }}Command) Run(args []string) int {
 		c.FlagSecrets,
 		c.FlagScrts,
 		func() {
-			opts = append(opts, {{ .Pkg }}.DefaultSecrets())
+			{{ if (and (eq .ResourceType "host-catalog") (eq .SubActionPrefix "plugin")) }} opts = append(opts, {{ .Pkg }}.DefaultSecrets()){{ end }}
 		},
 		func(in map[string]interface{}) {
-			opts = append(opts, {{ .Pkg }}.WithSecrets(in))
+			{{ if (and (eq .ResourceType "host-catalog") (eq .SubActionPrefix "plugin")) }}opts = append(opts, {{ .Pkg }}.WithSecrets(in)){{ end }}
+			{{ if (and (eq .ResourceType "credential") (eq .SubActionPrefix "kv")) }}opts = append(opts, {{ .Pkg }}.WithKvCredentialSecrets(in)){{ end }}
```

Comment:
In particular, it's not clear to me (yet) why this needs a special `WithKvCredentialSecrets`

Reply:
Went back and looked at what we do elsewhere and it seems like this should just be `WithSecrets`

Reply:
If we want to keep it typed, please break up the HasGenericSecrets in some way to compose what is needed -- don't put type checks in the templates

Reply:
The function WithKvCredentialSecrets is generated from the api package and it following the same naming patterns as the other credential types. For example: `WithSshPrivateKeyCredentialUsername`

Reply:
I replaced the string comparison conditionals with IsPluginType & IsKvCredentialType